delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/routes/StoragesRoutes.scala

```diff
@@ -164,33 +165,39 @@ final class StoragesRoutes(
                           },
                           // Deprecate a storage
                           (delete & parameter("rev".as[Long])) { rev =>
-                            authorizeFor(ref, permissions.write).apply {
+                            authorizeFor(ref, Write).apply {
                               emit(storages.deprecate(id, ref, rev).mapValue(_.metadata))
                             }
                           },
                           // Fetch a storage
-                          get {
-                            fetch(id, ref)
+                          (get & idSegmentRef(id) & authorizeFor(ref, Read)) { id =>
+                            emit(storages.fetch(id, ref).leftWiden[StorageRejection].rejectOn[StorageNotFound])
                           }
                         )
                       }
                     },
                     // Fetch a storage original source
-                    (pathPrefix("source") & get & pathEndOrSingleSlash) {
-                      operationName(s"$prefixSegment/storages/{org}/{project}/{id}/source") {
-                        fetchSource(id, ref, res => Storage.encryptSource(res.value.source, crypto).toOption.get)
+                    (pathPrefix("source") & get & pathEndOrSingleSlash & idSegmentRef(id)) { id =>
+                      authorizeFor(ref, Read).apply {
+                        operationName(s"$prefixSegment/storages/{org}/{project}/{id}/source") {
+                          val sourceIO = storages
+                            .fetch(id, ref)
+                            .map(res => Storage.encryptSource(res.value.source, crypto).toOption.get)
```

Comment:
while this was there before, `toOption.get` needs to be handled differently because it can throw without a descriptive message

Reply:
Not at this scope.

Plus in theory that does not happen, since we do check for encryption and decryption on storage evaluate method. That means that if a storage made it to the by it should be possible to encrypt/decrypt it.

It is true that after some change we did some weeks ago, now is possible to change the app.conf crypto and mess it all up. In any case, it is not on the scope of this PR to fix it.

Reply:
Notice that this is not only happening on the routes, it also happens here: 

- `EventSerializer.jsonSecretEncryptEncoder`
- `StorageEvent.jsonSecretEncryptEncoder`